Tests/UnitTests/TestHelpers/TestLogHandler.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  TestLogHandler.swift
+//
+//  Created by Nacho Soto on 8/19/22.
+
+@testable import RevenueCat
+
+/// Provides a `Logger.VerboseLogHandler` that wraps the default implementation
+/// and allows introspecting logged messages.
+///
+/// - Warning: this will implicitly override `Purchases.verboseLogHandler`.
+/// - Note: this only _wraps_ the original implementation, so messages are still logged to the console.
+///
+/// This type can be used with the RAII pattern.
+///
+/// ### Examples:
+/// - Initialize `TestLogHandler` for every test. This ensures that the lifetime
+/// matches that of the test, and the observed logged messages are those that happen during the test:
+///
+/// ```swift
+/// private var testLogHandler: TestLogHandler!
+/// override func setUp() {
+///     super.setUp()
+///     self.testLogHandler = TestLogHandler()
+/// }
+///
+/// override func tearDown() {
+///     self.testLogHandler = nil
+///     super.tearDown()
+/// }
+/// ```
+///
+/// - Alternatively, `TestLogHandler` can be used locally within a single test:
+///
+/// ```swift
+/// func testExample() {
+///     let logHandler = TestLogHandler()
+///
+///     // Run some code
+///
+///     expect(logHandler.loggedMessages.onlyElement?.message) == "Expected log"
+/// }
+/// ```
+final class TestLogHandler {
+
+    typealias MessageData = (level: LogLevel, message: String)
+
+    var messages: [MessageData] { return self.loggedMessages.value }
+
+    init() { Self.sharedHandler.add(observer: self) }
+
+    deinit { Self.sharedHandler.remove(observer: self) }
+
+    private let loggedMessages: Atomic<[MessageData]> = .init([])
+
+    private static let sharedHandler: SharedTestLogHandler = {
+        let handler = SharedTestLogHandler()
+        handler.install()
```

Comment:
Hmm personally, I don't like that an object contruction such as this has a side effect like this... Could we add a method on the `TestLogHandler` so it gets called only on the tests that require it instead? It might be more verbosity on the call side, but I think it makes it clearer.